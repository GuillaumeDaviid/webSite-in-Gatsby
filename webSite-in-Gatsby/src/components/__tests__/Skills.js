import React from "react"
import renderer from "react-test-renderer"
import Skills from "../Skills.tsx"
import { render, screen } from "@testing-library/react"

describe("Skills", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Skills />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should have a title element", () => {
    render(<Skills />)

    const title = screen.getByTestId('title');
    expect(title).toBeTruthy();
  })

  it("should have a project block", () => {
    render(<Skills />)

    const project = screen.getByTestId('project');
    expect(project).toBeTruthy();
  })

  it("should have a separator element", () => {
    render(<Skills />)

    const separator = screen.getByTestId('separator');
    expect(separator).toBeTruthy();
  })

  it("should have 12 skills elements", () => {
    render(<Skills />)

    const skills = screen.getAllByTestId('skills');
    expect(skills).toHaveLength(12);
  })
})