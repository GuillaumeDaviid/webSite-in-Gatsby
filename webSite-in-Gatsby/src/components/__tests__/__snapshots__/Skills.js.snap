// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills renders correctly 1`] = `
<div>
  <div
    className="project"
  >
    <div
      className="project_block"
      data-testid="project"
    >
      <a
        className="project_btn"
        target="_blank"
      >
        VOTRE PROJET
      </a>
    </div>
  </div>
  <div
    className="Skill"
  >
    <h2
      data-testid="title"
    >
      COMPETENCES
    </h2>
    <div
      className="Skill_separator"
      data-testid="separator"
    />
    <div
      className="Skill_skills"
    >
      <div
        className="skills"
        data-testid="skills"
      >
        HTML/CSS
      </div>
      <div
        className="skills"
        data-testid="skills"
      >
        SASS/SCSS
      </div>
      <div
        className="skills"
        data-testid="skills"
      >
        TypeScript
      </div>
      <div
        className="skills"
        data-testid="skills"
      >
        React.JS
      </div>
      <div
        className="skills"
        data-testid="skills"
      >
        NodeJS
      </div>
      <div
        className="skills"
        data-testid="skills"
      >
        NextJS
      </div>
      <div
        className="skills"
        data-testid="skills"
      >
        Javascript
      </div>
      <div
        className="skills"
        data-testid="skills"
      >
        SQL
      </div>
      <div
        className="skills"
        data-testid="skills"
      >
        PHP
      </div>
      <div
        className="skills"
        data-testid="skills"
      >
        Python
      </div>
      <div
        className="skills"
        data-testid="skills"
      >
        MySQL
      </div>
      <div
        className="skills"
        data-testid="skills"
      >
        Gatsby
      </div>
    </div>
  </div>
</div>
`;
